src/dhcp/server.c

```diff
@@ -456,7 +456,7 @@ static void init_packet(GDHCPServer *dhcp_server, struct dhcp_packet *packet,
 	packet->gateway_nip = client_packet->gateway_nip;
 	packet->ciaddr = client_packet->ciaddr;
 	dhcp_add_option_uint32(packet, DHCP_SERVER_ID,
-						dhcp_server->server_nip);
+						get_be32(&dhcp_server->server_nip));
```

Comment:
I think I have to modify my patch to remove the line you mentioned, after that, I think it will work. Oh, without this patch, I already test miracast with two machines running miracalcast, so, apply this patch just for increase compatibility among implementations.